### DISCUSSION TITLE
I can't use DDEV in Windows PowerShell after installing it in WSL2

### DISCUSSION THREAD

**Discussion by wcndave** [ddev/ddev#6793] [Q&A] [answered]
Following these guidelines: https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/#wsl2-docker-ce-inside-install-script
I have installed ddev, and from the ubuntu command prompt I can run both `ddev` and `docker` commands which return a usage message, so it seems everything is good to go.

However, when I run from powershell prompt, `ddev` command works (returns usage), but` ddev version` doesn't 
it returns:
```
PS C:\Users\Dave\Documents> ddev version
time="2024-11-30T18:25:09+01:00" level=fatal msg="\x1b[31mUnable to get Docker context: 
unable to run 'docker context inspect' - please make sure Docker client is in path and up-to-date: 
exec: \"docker\": executable file not found in %PATH%\x1b[0m"
```
I am not sure what I am missing from the instructions as I think I followed them all step by step....
Can someone advise me?  Thanks!

**Comment by rfay** [comment, marked as answer]
Hi @wcndave - 

When you install DDEV with docker-ce (glad you did that) you will use DDEV **inside the WSL2 distro** (typically Ubuntu). You won't be using it in PowerShell.

Please go back and read the [Important considerations for DDEV and WSL2](https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/#important-considerations-for-wsl2-and-ddev)

> Execute DDEV commands inside WSL2.
> You’ll want to run DDEV commands inside Ubuntu, for example, and never on the Windows side in PowerShell or Git Bash.

> Projects should live under the home directory of the Linux filesystem.
> WSL2’s Linux filesystem (e.g. /home/<your_username>) is much faster and has proper permissions, so keep your projects there and not in the slower Windows filesystem (/mnt/c).

**Reply by wcndave** [reply]
Ah right, the reason I had not done this, was because 1) everything I read about ddev said it's so you can run linx commands etc without needing to be IN a linux environment, but the more "familiar" windows one. and 2) I ran the script the first time, and ddev returned: 

```
dave@Dave-Main-2020:~$ ddev --version
Command 'ddev' not found, did you mean:
  command 'ddv' from deb ncbi-tools-x11 (6.1.20170106+dfsg1-10)
Try: sudo apt install <deb name>
```

Whereas it DID return instructions when run in powershell....

However, having done a second time, I can see that the script didn't work - I ran through each command manually one at a time, and was able to install and resolve everything that came up and THEN it runs ok inside the distro.

So I think a healthy warning that running the script might not work, and if "ddev --version" inside the distro doesn't return expected results to go through and manually run each command in the script would be good.

Many thanks - hopefully I can get it going from here!
Dave

**Reply by wcndave** [reply]
Sadly not...

```
dave@Dave-Main-2020:~$ mkdir ddev-drupal-test1
dave@Dave-Main-2020:~$ cd ddev-drupal-test1/
dave@Dave-Main-2020:~/ddev-drupal-test1$ ddev config --project-type=drupal --php-version=8.3 --docroot=web
Creating a new DDEV project config in the current directory (/home/dave/ddev-drupal-test1)
Once completed, your configuration will be written to /home/dave/ddev-drupal-test1/.ddev/config.yaml

Created docroot directory at /home/dave/ddev-drupal-test1/web
Configuring a 'drupal' project named 'ddev-drupal-test1' with docroot 'web' at '/home/dave/ddev-drupal-test1/web'.
For full details use 'ddev describe'.
Unable to detect Drupal version, continuing
No settings.php file exists, creating one
Configuration complete. You may now run 'ddev start'.
dave@Dave-Main-2020:~/ddev-drupal-test1$ ddev start
Network ddev_default created
It looks like you have a new DDEV release.
May we send anonymous DDEV usage statistics and errors?
To know what we will see please take a look at
https://ddev.readthedocs.io/en/stable/users/usage/diagnostics/#opt-in-usage-information
Permission to beam up? [Y/n] (yes): n
Starting ddev-drupal-test1...
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 127.0.0.53:53: server misbehaving
Unable to pull Docker images: exit status 1
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 127.0.0.53:53: server misbehaving
Unable to determine Docker space usage:
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 127.0.0.53:53: server misbehaving
Failed to start ddev-drupal-test1: failed to RunSimpleContainer to chown volumes: failed to pull image ddev/ddev-webserver:v1.23.5: exit status 1, output=
dave@Dave-Main-2020:~/ddev-drupal-test1$
```

**Reply by rfay** [reply]
Congrats on the progress! Now your problem seems to be that you don't have internet connectivity in WSL2, or registry-1.docker.io was having a bad day. You can try `docker pull busybox` and when that works these things should work; this has nothing to do with DDEV.

If you *do* prefer to use DDEV on the Windows side (with PowerShell) the instructions for that are on that same page, https://ddev.readthedocs.io/en/stable/users/install/ddev-installation/#traditional-windows

**Reply by wcndave** [reply]
it's ok, just updated resolve.conf
I've found that a few times, the networking aspect of VMs was often the tricky part, and not having a good nameserver out of the box does seem a bit odd - but moving forward nonetheless!  Thanks!
